lib/usecases/clear_cached_surveys_use_case.dart

```diff
@@ -0,0 +1,18 @@
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:survey_flutter/api/storage/hive_storage.dart';
+import 'package:survey_flutter/usecases/base/base_use_case.dart';
+
+final clearCachedSurveysUseCaseProvider = Provider(
+    (ref) => ClearCachedSurveysUseCase(ref.watch(hiveStorageProvider)));
+
+class ClearCachedSurveysUseCase extends NoParamsUseCase<int> {
+  final HiveStorage _hiveStorage;
+
+  ClearCachedSurveysUseCase(this._hiveStorage);
+
+  @override
+  Future<Result<int>> call() async {
```

Comment:
we don't use the result so it's better to return `void`
```suggestion
class ClearCachedSurveysUseCase extends NoParamsUseCase<void> {
  final HiveStorage _hiveStorage;

  ClearCachedSurveysUseCase(this._hiveStorage);

  @override
  Future<Result<void>> call() async {
```

Reply:
@manh-t I tried to return `void` at first. But it cannot be unit tested. I researched a little and I found one website that says we should avoid returning `void` as much as we could.